### PR TITLE
[EPM pallet]: remove `number of signed submissions`

### PR DESCRIPTION
**Pull request by niklasad1**
Closing #9229

polkadot companion: https://github.com/paritytech/polkadot/pull/5012

### CODE REVIEW

File: frame/election-provider-multi-phase/src/lib.rs [hunk 1]
```diff
@@ -967,17 +967,16 @@ pub mod pallet {
...
 			origin: OriginFor<T>,
 			raw_solution: Box<RawSolution<SolutionOf<T>>>,
-			num_signed_submissions: u32,
 		) -> DispatchResult {
 			let who = ensure_signed(origin)?;
 
 			// ensure witness data is correct.
 			ensure!(
-				num_signed_submissions >=
+				T::SignedMaxSubmissions::get() >=
```
Comment (niklasad1) [inline frame/election-provider-multi-phase/src/lib.rs:979]: not sure if this is needed anymore?!

File: frame/election-provider-multi-phase/src/lib.rs [hunk 2]
```diff
@@ -967,17 +967,16 @@ pub mod pallet {
 		/// # <weight>
 		/// Queue size must be provided as witness data.
 		/// # </weight>
-		#[pallet::weight(T::WeightInfo::submit(*num_signed_submissions))]
+		#[pallet::weight(T::DbWeight::get().reads_writes(5, 3))]
```
Comment (niklasad1) [inline frame/election-provider-multi-phase/src/lib.rs:970]: https://github.com/paritytech/polkadot/blob/master/runtime/polkadot/src/weights/pallet_election_provider_multi_phase.rs#L117-#L118

File: frame/election-provider-multi-phase/src/lib.rs [hunk 3]
```diff
@@ -1970,11 +1961,9 @@ mod tests {
 					score: ElectionScore { minimal_stake: (5 + s).into(), ..Default::default() },
 					..Default::default()
 				};
-				assert_ok!(MultiPhase::submit(
-					crate::mock::Origin::signed(99),
-					Box::new(solution),
-					MultiPhase::signed_submissions().len() as u32
-				));
+				assert_ok!(
+					MultiPhase::submit(crate::mock::Origin::signed(99), Box::new(solution),)
```
Comment (niklasad1) [inline frame/election-provider-multi-phase/src/lib.rs:1965]:
```suggestion
					MultiPhase::submit(crate::mock::Origin::signed(99), Box::new(solution))
```

File: frame/election-provider-multi-phase/src/signed.rs [hunk 4]
```diff
@@ -540,12 +536,6 @@ mod tests {
 			// submit this once correctly
 			assert_ok!(submit_with_witness(Origin::signed(99), solution.clone()));
 			assert_eq!(MultiPhase::signed_submissions().len(), 1);
-
-			// now try and cheat by passing a lower queue length
-			assert_noop!(
-				MultiPhase::submit(Origin::signed(99), Box::new(solution), 0),
```
Comment (niklasad1) [inline frame/election-provider-multi-phase/src/signed.rs:546]: this is not possible anymore so I removed this assertion.

File: frame/election-provider-multi-phase/src/lib.rs (hunk 2 above)
Comment (kianenigma) [reply frame/election-provider-multi-phase/src/lib.rs:970]:
```suggestion
		#[pallet::weight(T::WeightInfo::submit())]
```

File: frame/election-provider-multi-phase/src/lib.rs [hunk 5]
```diff
@@ -963,25 +963,13 @@ pub mod pallet {
 		///
 		/// A deposit is reserved and recorded for the solution. Based on the outcome, the solution
 		/// might be rewarded, slashed, or get all or a part of the deposit back.
-		///
-		/// # <weight>
-		/// Queue size must be provided as witness data.
-		/// # </weight>
-		#[pallet::weight(T::WeightInfo::submit(*num_signed_submissions))]
+		#[pallet::weight(T::WeightInfo::submit(0))]
```
Comment (kianenigma) [inline frame/election-provider-multi-phase/src/lib.rs:966]:
```suggestion
		#[pallet::weight(T::WeightInfo::submit())]
```

There needs to be no argument, you need to change the signature in `trait WeightInfo`.

**Review by kianenigma — APPROVED** [review]
expcet: 

1. https://github.com/paritytech/substrate/pull/10945/files#r818693002
2. need to run the bench bot again.

**Comment by kianenigma** [comment]
/benchmark runtime pallet pallet_election_provider_multi_phase

**Comment by emostov** [comment]
lgtm - the polkadot companion just needs a few more updates, see https://github.com/paritytech/polkadot/pull/5012#issuecomment-1058354485

**Comment by niklasad1** [comment]
@emostov can you take another look? Hopefully good to go now :)

**Review by emostov — APPROVED** [review]

**Pull request merged**